packages/web/app/features/ProgramRegistry/PatientProgramRegistryUpdateFormModal.jsx

```diff
@@ -0,0 +1,435 @@
+  return yup.object().shape({
+    clinicalStatusId: optionalForeignKey(),
+    conditions: yup.object().shape({
+      confirmedSection: yup.array().of(getConditionShape(getTranslation)),
+      resolvedSection: yup.array().of(getConditionShape(getTranslation)),
+      recordedInErrorSection: yup.array().of(getConditionShape(getTranslation)),
+    }),
+  });
+};
+
+const useUpdateProgramRegistryMutation = (patientId, registrationId) => {
+  const api = useApi();
+  const queryClient = useQueryClient();
+
+  return useMutation(
+    data => {
+      return api.put(`patient/programRegistration/${registrationId}`, data);
+    },
+    {
+      onSuccess: () => {
+        queryClient.invalidateQueries([`infoPaneListItem-${PANE_SECTION_IDS.PROGRAM_REGISTRY}`]);
+        queryClient.invalidateQueries(['patient', patientId]);
+      },
+    },
+  );
+};
+
+const getGroupedData = rows => {
+  const groupMapping = {
+    confirmedSection: [
+      'suspected',
+      'underInvestigation',
+      'confirmed',
+      'unknown',
+      'inRemission',
+      'notApplicable',
+    ],
+    resolvedSection: ['disproven', 'resolved'],
+    recordedInErrorSection: ['recordedInError'],
+  };
+
+  // Initialize result object
+  const groupedData = { confirmedSection: [{}], resolvedSection: [], recordedInErrorSection: [] };
+
+  // Process rows
+  rows.forEach(({ id, conditionCategory, date, programRegistryCondition, reasonForChange }) => {
+    for (const [group, conditions] of Object.entries(groupMapping)) {
+      if (conditions.includes(conditionCategory)) {
+        groupedData[group].push({
+          id,
+          conditionId: programRegistryCondition.id,
+          name: programRegistryCondition.name,
+          date,
+          conditionCategory,
+          reasonForChange,
+        });
+        break;
+      }
+    }
+  });
+  Object.keys(groupedData).forEach(group => {
+    groupedData[group].sort((a, b) => a.name?.localeCompare(b?.name));
+  });
+  return groupedData;
+};
+
+export const PatientProgramRegistryUpdateFormModal = ({
+  patientProgramRegistration = {},
+  onClose,
+  open,
+}) => {
+  const {
+    id: registrationId,
+    programRegistryId,
+    patientId,
+    clinicalStatusId,
+  } = patientProgramRegistration;
+  const { getTranslation } = useTranslation();
+  const { mutateAsync: submit, isLoading: isSubmitting } = useUpdateProgramRegistryMutation(
+    patientId,
+    registrationId,
+  );
+  const { data: conditions = [], isLoading } = usePatientProgramRegistryConditionsQuery(
+    patientId,
+    programRegistryId,
+  );
+  const programRegistryStatusSuggester = useSuggester('programRegistryClinicalStatus', {
+    baseQueryParameters: { programRegistryId },
+  });
+
+  const handleSubmit = async data => {
+    // flatten data for form submission
+    const conditions = Object.values(data.conditions)
+      .reduce((acc, group) => acc.concat(group), [])
+      .filter(({ conditionId }) => conditionId);
```

Comment:
A thing you can do is this 2
```suggestion
    const conditions = Object.values(data.conditions).flatMap(group => group)
      .filter(({ conditionId }) => conditionId);
```